src/main/java/de/berlin/lostberlin/controller/OrderController.java

```diff
@@ -1,85 +1,106 @@
 package de.berlin.lostberlin.controller;
 
-import de.berlin.lostberlin.exception.ResourceNotFoundException;
-import de.berlin.lostberlin.model.Order;
+import de.berlin.lostberlin.model.order.client.OrderFullDao;
+import de.berlin.lostberlin.model.order.client.OrderPostDto;
+import de.berlin.lostberlin.model.order.client.OrderShortDao;
+import de.berlin.lostberlin.model.order.client.OrderStatusDao;
+import de.berlin.lostberlin.model.order.persistence.Order;
+import de.berlin.lostberlin.model.order.persistence.StatusTypes;
 import de.berlin.lostberlin.repository.OrderRepository;
+import de.berlin.lostberlin.service.OrderService;
 import de.berlin.lostberlin.service.mail.MailService;
-import org.apache.commons.validator.routines.EmailValidator;
-import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
+import org.springframework.web.server.ResponseStatusException;
+
 import javax.validation.Valid;
 
 import java.util.List;
 
 @RestController
+@RequestMapping("/orders")
 public class OrderController {
-
-    @Autowired
     private OrderRepository orderRepo;
-
-    @Autowired
     private MailService mailService;
+    private OrderService orderService;
 
-    @GetMapping("/orders/all")
-    public List<Order> getAllOrders() {
-        return orderRepo.findAll();
+    public OrderController(OrderRepository orderRepo,MailService mailService, OrderService orderService) {
+        this.orderRepo = orderRepo;
+        this.mailService = mailService;
+        this.orderService = orderService;
     }
 
-    @GetMapping("/orders")
+    @GetMapping("/all")
+    public List<Order> getAllOrders() {
+        return orderService.retrieveAllOrders();
+    }
 
-    public List<Order> getOrders(@RequestParam (required=true, value = "business_id") Long businessId) {
+    @GetMapping
+    public List<OrderShortDao> getOrders(@RequestParam(value = "business_id") Long businessId) {
+        return orderService.retrieveOrdersByBusinessId(businessId);
+    }
 
-        return orderRepo.findAllByBusinessId(businessId);
+    @GetMapping("/{order_number}")
```

Comment:
We should make it uniform: all methods should return ResponseEntity.

Reply:
I created a separate branch for order service and controller refactoring